rider/src/main/kotlin/com/jetbrains/rider/plugins/unity/notifications/UnityAutoSaveConfigureNotification.kt

```diff
@@ -0,0 +1,128 @@
+import com.intellij.ide.util.PropertiesComponent
+import com.intellij.openapi.editor.Editor
+import com.intellij.openapi.editor.EditorFactory
+import com.intellij.openapi.editor.event.DocumentEvent
+import com.intellij.openapi.editor.event.DocumentListener
+import com.intellij.openapi.editor.impl.EditorImpl
+import com.intellij.openapi.fileEditor.FileEditorManager
+import com.intellij.openapi.fileEditor.impl.text.TextEditorProvider
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.util.Key
+import com.intellij.ui.EditorNotificationPanel
+import com.intellij.ui.LightColors
+import com.intellij.util.SingleAlarm
+import com.jetbrains.rd.util.lifetime.Lifetime
+import com.jetbrains.rd.util.lifetime.isAlive
+import com.jetbrains.rd.util.lifetime.onTermination
+import com.jetbrains.rd.util.reactive.*
+import com.jetbrains.rdclient.util.idea.LifetimedProjectComponent
+import com.jetbrains.rider.UnityProjectDiscoverer
+import com.jetbrains.rider.document.getFirstEditor
+import com.jetbrains.rider.model.EditorState
+import com.jetbrains.rider.model.ScriptCompilationDuringPlay
+import com.jetbrains.rider.plugins.unity.UnityHost
+import com.jetbrains.rider.projectView.SolutionLifecycleHost
+import com.jetbrains.rider.util.idea.application
+
+class UnityAutoSaveConfigureNotification (project: Project, private val unityProjectDiscoverer: UnityProjectDiscoverer,
+                                          private val unityHost: UnityHost, solutionLifecycleHost: SolutionLifecycleHost) : LifetimedProjectComponent(project){
+
+    private val propertiesComponent: PropertiesComponent = PropertiesComponent.getInstance()
+    private var lifetimeDefinition = componentLifetime.createNested()
+    private val KEY = Key.create<Any>("PromoteAutoSave")
+
+    companion object {
+        private const val settingName = "do_not_show_unity_auto_save_notification"
+    }
+
+    init {
+        solutionLifecycleHost.isBackendLoaded.whenTrue(componentLifetime) {
+            if (!propertiesComponent.getBoolean(settingName) && unityProjectDiscoverer.isUnityProject) {
+
+                val eventMulticaster = EditorFactory.getInstance().eventMulticaster
+                val documentListener: DocumentListener = object : DocumentListener {
+                    override fun documentChanged(event: DocumentEvent) {
+                        if (unityHost.model.editorState.valueOrDefault(EditorState.Disconnected) != EditorState.ConnectedPlay)
+                            return
+
+                        if (!unityHost.model.scriptCompilationDuringPlay.hasValue)
+                            return
+
+                        if (unityHost.model.scriptCompilationDuringPlay.valueOrThrow != ScriptCompilationDuringPlay.RecompileAndContinuePlaying)
+                            return
+
+                        if (!lifetimeDefinition.isAlive)
+                            return
+
+                        val generalSettings = GeneralSettings.getInstance()
```

Comment:
let's move GeneralSettings.getInstance out of this scope, so it won't be calculated for every doc change

Reply:
Ooops. Thanks!